readers/hdfs/src/main/java/com/criteo/hadoop/garmadon/hdfs/offset/HdfsOffsetComputer.java

```diff
@@ -45,51 +50,107 @@ public HdfsOffsetComputer(FileSystem fs, Path basePath, @Nullable String kafkaCl
+                        if (isPartitionComputedByThisReader(partitionId, partitionIdsSet) &&
+                            chekCurrentFileIndexBiggerTheOneInHashPartitionDay(partitionId, listedDay, index, resultFile)) {
+                            HashMap<String, FinalEventPartitionFile> dateFinalEventPartitionFile =
+                                (HashMap<String, FinalEventPartitionFile>) resultFile.computeIfAbsent(partitionId, HashMap::new);
+                            dateFinalEventPartitionFile.put(listedDay,
+                                new FinalEventPartitionFile(index, status.getPath()));
+                            resultFile.put(partitionId, dateFinalEventPartitionFile);
+                        }
+                    } catch (NumberFormatException e) {
+                        LOGGER.info("Couldn't deviate a valid offset from '{}'", fileName);
                     }
-                } catch (NumberFormatException e) {
-                    LOGGER.info("Couldn't deviate a valid offset from '{}'", fileName);
                 }
             }
         }
 
+        // Get last offset
+        for (Map.Entry<Integer, Map<String, FinalEventPartitionFile>> partitionIdDateFile : resultFile.entrySet()) {
+            result.put(partitionIdDateFile.getKey(), getMaxOffset(partitionIdDateFile.getValue()));
+        }
+
         return result;
     }
 
-    private String computeDirNamesPattern(LocalDateTime today, int backlogDays) {
-        List<String> dirnames = new ArrayList<>(backlogDays);
+    private boolean isPartitionComputedByThisReader(int partitionId, Set<Integer> partitionIdsSet) {
+        return partitionIdsSet.contains(partitionId);
+    }
+
+    private boolean chekCurrentFileIndexBiggerTheOneInHashPartitionDay(int partitionId, String listedDay, long index, Map<Integer,
+        Map<String, FinalEventPartitionFile>> resultFile) {
+        return isPartitionDayNotAlreadyInHashPartitionDay(partitionId, listedDay, resultFile) || index > resultFile.get(partitionId).get(listedDay).getIndex();
+    }
 
-        for (int i = 0; i < backlogDays; ++i) {
-            LocalDateTime day = today.minusDays(i);
+    private boolean isPartitionDayNotAlreadyInHashPartitionDay(int partitionId, String listedDay, Map<Integer,
+        Map<String, FinalEventPartitionFile>> resultFile) {
+        return !(resultFile.containsKey(partitionId) && resultFile.get(partitionId).containsKey(listedDay));
+    }
 
-            dirnames.add(computeDirName(day));
-        }
+    protected Long getMaxOffset(Map<String, FinalEventPartitionFile> dateFinalEventPartitionFile) {
+        List<Long> maxOffsets = new ArrayList<>();
+        dateFinalEventPartitionFile.values().stream()
+            .forEach(finalEventPartitionFile -> {
+                // Read parquet file
+                ParquetFileReader pFR;
+                try {
+                    pFR = ParquetFileReader.open(fs.getConf(), finalEventPartitionFile.getFilePath());
```

Comment:
it looks like pFR is never closed. It is safer to use try-with-resource

 try (ParquetFileReader pFR = ParquetFileReader.open(fs.getConf(), finalEventPartitionFile.getFilePath())) {
   //stream processing
} catch [...]